### PR TITLE
poll data refactoring to graphql

### PR DESCRIPTION
**Pull request by plehocky**
| Q                        | A 
| ------------------------ | ---
| Fixed Issue?             | `FD-554` 
| Tests Added + Pass?      | Yes
| Any Dependency Changes?  |

https://frinxhelpdesk.atlassian.net/jira/software/c/projects/FD/boards/70?modal=detail&selectedIssue=FD-554
refactored pollData to use graphql and added extended filtering options

### CODE REVIEW

File: packages/frinx-workflow-ui/src/pages/poll-data/poll-data.tsx [hunk 1]
```diff
@@ -11,80 +18,228 @@ import {
...
+
+const PollDataPage = () => {
+  const [orderBy, setOrderBy] = useState<OrderBy>({ sortKey: 'lastPollTime', direction: 'asc' });
+  const [filter, setFilter] = useState<Filter>({});
+  const [inputs, setInputs] = useState({
+    queueName: '',
+    workerId: '',
+    domain: '',
+    beforeDate: '' as string | undefined,
+    afterDate: '' as string | undefined,
```
Comment (MarcoMruz) [inline packages/frinx-workflow-ui/src/pages/poll-data/poll-data.tsx:84]: we do not use type assertions just to satisfy TS

File: packages/frinx-workflow-ui/src/pages/poll-data/poll-data.tsx [hunk 2]
```diff
@@ -11,80 +18,228 @@ import {
...
+      }
     }
-    return worfklowName.includes(query);
+  }
+`;
+
+const PollDataPage = () => {
+  const [orderBy, setOrderBy] = useState<OrderBy>({ sortKey: 'lastPollTime', direction: 'asc' });
+  const [filter, setFilter] = useState<Filter>({});
+  const [inputs, setInputs] = useState({
```
Comment (MarcoMruz) [inline packages/frinx-workflow-ui/src/pages/poll-data/poll-data.tsx:79]: if you are using object for a state cleaner way would be to use type which will define what properties are we using and also if we will want to use this kind of object somewhere else

File: packages/frinx-workflow-ui/src/pages/poll-data/poll-data.tsx [hunk 3]
```diff
@@ -11,80 +18,228 @@ import {
...
+  function getISODateString(dateString: string | undefined): string | undefined {
+    if (dateString === undefined) {
+      return undefined;
+    }
+    const dateObj = new Date(dateString);
+    if (Number.isNaN(dateObj.getTime())) {
+      return undefined;
+    }
+    return dateObj.toISOString();
+  }
```
Comment (MarcoMruz) [inline packages/frinx-workflow-ui/src/pages/poll-data/poll-data.tsx:114]: function does not have to be in component move it outside or to helpers/utils file

File: packages/frinx-workflow-ui/src/pages/poll-data/poll-data.tsx [hunk 4]
```diff
@@ -11,80 +18,228 @@ import {
...
+      ...inputs,
+      beforeDate: getISODateString(inputs.beforeDate),
+      afterDate: getISODateString(inputs.afterDate),
+    });
+  };
 
-    setItemList(results);
-  }, [keywords, data, setItemList]);
+  const resetFilter = () => {
+    setFilter({});
```
Comment (MarcoMruz) [inline packages/frinx-workflow-ui/src/pages/poll-data/poll-data.tsx:125]: if you are reseting value or object will not have any properties then set the value to null

**Review by MarcoMruz — CHANGES_REQUESTED** [review]

**Review by soson — APPROVED** [review]

**Review by soson — APPROVED** [review]

**Review by MarcoMruz — APPROVED** [review]

**Pull request merged**